src/proplists.ts

```diff
@@ -11,9 +12,12 @@ export function proplistToDict<T>(proplist: Proplist<T>): Dict<T> {
   }, {});
 }
 
-export function dictToProplist<T>(dict: Dict<T>): Proplist<T> {
+export function dictToProplist<T>(dict: Dict<T>): MutableProplist<T> {
   return Object.keys(dict).reduce(
-    (memo: Proplist<T>, key): Proplist<T> => [...memo, [key, dict[key]]],
+    (memo: MutableProplist<T>, key): MutableProplist<T> => [
```

Comment:
So, as we're returning a copy, the output can theoretically be mutable to consumers, as we don't maintain a reference to it so don't care if it changes.  The only time we need to care is when taking it as an argument, where we guarantee we're not going to change it.

In library types, it's better to take immutables and return mutables, because immutable slots can receive mutable types, but the reverse isn't true.

Reply:
So, this `memo: MutableProplist<T>`, could be immutable. Since we're not mutating it, andthis is never exposed.

Reply:
Resolved in conversation via Slack.